lib/src/serializable.dart

```diff
@@ -0,0 +1,237 @@
+// Copyright 2016 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+library serializable_module.src.serializable;
+
+@MirrorsUsed(metaTargets: 'serializable_module.src.serializable.Reflectable')
+import 'dart:mirrors';
+import 'dart:async';
+import 'dart:html' show window, CustomEvent;
+import 'dart:js' show JsObject, context;
+
+import 'package:logging/logging.dart';
+import 'package:w_common/w_common.dart' show JsonSerializable;
+
+import 'event.dart';
+import 'module.dart';
+
+// Any classes / methods that are going to be reflected must annotate with this
+class Reflectable {
+  const Reflectable();
+}
+
+abstract class Bridge<T> {
+  StreamController<Map> _eventController;
+
+  Bridge() {
+    _eventController = new StreamController<Map>.broadcast();
+  }
+
+  void broadcast(Map dataToSend);
+  void handleEvent(T apiCall);
+
+  Stream<Map> get eventReceived => _eventController.stream;
+}
+
+// This class is used to communicate across a WKWebView in an iOS app
+class WKWebViewBridge extends Bridge<CustomEvent> {
+  WKWebViewBridge() {
+    window.on['bridge'].listen(handleEvent);
+  }
+
+  @override
+  void broadcast(Map dataToSend) {
+    wkBridge?.callMethod('postMessage', [new JsObject.jsify(dataToSend)]);
+  }
+
+  @override
+  void handleEvent(CustomEvent apiCall) {
+    _eventController.add(apiCall.detail);
+  }
+
+  JsObject get wkBridge {
+    if (context['webkit'] != null) {
+      return context['webkit']['messageHandlers']['bridge'];
+    } else {
+      return null;
+    }
+  }
+}
+
+class SerializableEvent<T> extends Event<T> {
+  SerializableEvent(this._eventKey, DispatchKey dispatchKey)
+      : super(dispatchKey);
+
+  String _eventKey;
+  String get eventKey => _eventKey;
```

Comment:
That is a problem we will have to tackle in the future, being able to register a unique instance with bridge. When I talked with Trent initially we decided for now it is ok to just be able to handle one instance of a given module at a time.